--- config_unix.py~	2008-05-22 09:55:26.000000000 -0700
+++ config_unix.py	2008-05-22 09:55:43.000000000 -0700
@@ -11,12 +11,13 @@
 #these get prefixes with '/usr' and '/usr/local' or the $LOCALBASE
 origincdirs = ['/include', '/include/SDL', '/include/SDL11',
                '/include/smpeg' ]
-origlibdirs = ['/lib']
+origlibdirs = ['/lib','/lib64']
 
 
 
 
 def confirm(message):
+    return 1
     "ask a yes/no question, return result"
     reply = raw_input('\n' + message + ' [Y/n]:')
     if reply and string.lower(reply[0]) == 'n':
